src/js/api/shipment.js

```diff
@@ -0,0 +1,359 @@
+ *    - 'name' - Customer's contact name.
+ *    - 'phone' - Customer's contact phone.
+ *    - 'address' - Customer's address.
+ *        - 'line' - Customer's address line.
+ *        - 'city' - Customer's address city.
+ *        - 'postal_code' - Customer's postal code.
+ *        - 'state_code' - Customer's state province code.
+ *        - 'country_code' - Customer's 'ISO 3166-2' country code.
+ *  - 'attachments' - A list of RIPE Core attachment IDs.
+ * @param {Function} callback Function with the result of the request.
+ * @returns {XMLHttpRequest} The XMLHttpRequest instance of the API request.
+ */
+ripe.Ripe.prototype.updateShipment = function(number, options, callback) {
+    callback = typeof options === "function" ? options : callback;
+    options = typeof options === "function" || options === undefined ? {} : options;
+    const url = `${this.url}shipments/${number}`;
+    options = Object.assign(options, {
+        url: url,
+        method: "PUT",
+        params: options,
+        auth: true
+    });
+    options = this._build(options);
+    return this._cacheURL(options.url, options, callback);
+};
+
+/**
+ * Updates a shipment on Ripe Core.
+ *
+ * @param {Number} number The number of the shipment to find by.
+ * @param {Object} options An object with options, such as:
+ *  - 'status' - The shipment status.
+ *  - 'order' - The order associated with this shipment, if any.
+ *  - 'bulk_order' - The bulk order associated with this shipment, if any.
+ *  - 'description' - A brief description of this shipment.
+ *  - 'weight' - Weight of the package being shipped.
+ *  - 'weight_units' - Weight units (e.g. "kilograms" or "pounds").
+ *  - 'courier' - The courier for this shipment.
+ *  - 'service' - The transportation service used.
+ *  - 'package' - The package used.
+ *  - 'tracking_number' - The tracking number associated with this shipment.
+ *  - 'tracking_url' - The tracking URL associated with this shipment.
+ *  - 'carbon_neutral' - If this shipment should be carbon neutral.
+ *  - 'access_point_delivery' - If this shipment should be sent to an access point (e.g. "no", "optional" or "mandatory").
+ *  - 'pickup' - Pickup information.
+ *    - 'offset' - Time offset for pickup scheduling relative to shipment creation (millis).
+ *    - 'ready_time' - Opening hours for shipper facility (millis).
+ *    - 'close_time' - Closing hours for shipper facility (millis).
+ *  - 'shipping_date' - The date the shipment ended.
+ *  - 'delivery_date' - The date the shipment began.
+ *  - 'shipper' - Shipper's information.
+ *    - 'name' - Shipper's contact name.
+ *    - 'phone' - Shipper's contact phone.
+ *    - 'address' - Shipper's address.
+ *        - 'line' - Shipper's address line.
+ *        - 'city' - Shipper's address city.
+ *        - 'postal_code' - Shipper's postal code.
+ *        - 'state_code' - Shipper's state province code.
+ *        - 'country_code' - Shipper's 'ISO 3166-2' country code.

```

Comment:
I don't know if you should have this here but leave it vague in the data model in core. This is weird, since you expect a certain structure but have no way of verifying it/forcing it in the data model.

(this is just a comment, no change required)

Reply:
I understand. My thought process was that this needs to be documented somewhere. For example, Marina as a stakeholder of drop shipping constantly asks me what the format needs to be for certain fields. I thought about documenting it here in the SDK since it will be used by external teams or developers. This way it is documented but it can of course be moved somewhere else accordingly. What do you think @hugo-gomes ?